include/drivers/clock_control.h

```diff
@@ -139,25 +154,33 @@ static inline int clock_control_off(struct device *dev,
 	const struct clock_control_driver_api *api =
 		(const struct clock_control_driver_api *)dev->driver_api;
 
+	if (api->off == NULL) {
+		return -ENOTSUP;
+	}
+
 	return api->off(dev, sys);
 }
 
 /**
- * @brief Request clock to start with notification when clock has been started.
+ * @brief Asynchronously start clock resource with notification when clock has
+ *	  been started.
  *
  * Function is non-blocking and can be called from any context.
  * When clock is already running user callback will be called from the context
  * of the function call else it is called from other context (e.g. clock
  * interrupt).
  *
+ * @note It is not recommended to mix on/off API with request/release API.
+ *	 Particularly, that driver may implement just one approach.
+ *
  * @param dev 	   Device.
  * @param sys	   A pointer to an opaque data representing the sub-system.
  * @param data	   Data structure containing a callback that is called when
  *		   action is performed. Structure content must be valid until
  *		   clock is started and user callback is called. Can be NULL.
```

Comment:
Please document that a null `data` parameter means the operation will proceed without any way for the caller to detect when it completes or whether it succeeds.  (Assuming that is the promised behavior; right now it's not clear what promises are made.)